src/admin.ts

```diff
@@ -170,49 +106,22 @@ export class Admin {
    * @param username - The username for the new user
    * @param password - An optional password for the new user
    * @param options - Optional settings for the command
-   * @param callback - An optional callback, a Promise will be returned if none is provided
    */
-  addUser(username: string): Promise<Document>;
-  addUser(username: string, password: string): Promise<Document>;
-  addUser(username: string, options: AddUserOptions): Promise<Document>;
-  addUser(username: string, password: string, options: AddUserOptions): Promise<Document>;
-  /** @deprecated Callbacks are deprecated and will be removed in the next major version. See [mongodb-legacy](https://github.com/mongodb-js/nodejs-mongodb-legacy) for migration assistance */
-  addUser(username: string, callback: Callback<Document>): void;
-  /** @deprecated Callbacks are deprecated and will be removed in the next major version. See [mongodb-legacy](https://github.com/mongodb-js/nodejs-mongodb-legacy) for migration assistance */
-  addUser(username: string, password: string, callback: Callback<Document>): void;
-  /** @deprecated Callbacks are deprecated and will be removed in the next major version. See [mongodb-legacy](https://github.com/mongodb-js/nodejs-mongodb-legacy) for migration assistance */
-  addUser(username: string, options: AddUserOptions, callback: Callback<Document>): void;
-  /** @deprecated Callbacks are deprecated and will be removed in the next major version. See [mongodb-legacy](https://github.com/mongodb-js/nodejs-mongodb-legacy) for migration assistance */
-  addUser(
-    username: string,
-    password: string,
-    options: AddUserOptions,
-    callback: Callback<Document>
-  ): void;
-  addUser(
+  async addUser(
     username: string,
-    password?: string | AddUserOptions | Callback<Document>,
-    options?: AddUserOptions | Callback<Document>,
-    callback?: Callback<Document>
-  ): Promise<Document> | void {
-    if (typeof password === 'function') {
-      (callback = password), (password = undefined), (options = {});
-    } else if (typeof password !== 'string') {
-      if (typeof options === 'function') {
-        (callback = options), (options = password), (password = undefined);
-      } else {
-        (options = password), (callback = undefined), (password = undefined);
-      }
-    } else {
-      if (typeof options === 'function') (callback = options), (options = {});
-    }
-
-    options = Object.assign({ dbName: 'admin' }, options);
-
+    password?: string | AddUserOptions,
+    options?: AddUserOptions
+  ): Promise<Document> {
+    options =
+      options != null && typeof options === 'object'
```

Comment:
we removed the overload that documents this, so until we are ready to break the behavior, we should keep those overloads

Reply:
We can still drop the overload, and update the signature. Overloads are generally more useful when the return type will change based on a change in input, the DX of a string unioned with options object should be equivalent to what was here before.

Happy to bring it back tho, just my 2c